java-bootcamp-2018-tdd/src/main/java/com/globant/java_bootcamp_2018_tdd/RecentFileList.java

```diff
@@ -0,0 +1,66 @@
+package com.globant.java_bootcamp_2018_tdd;
+
+import java.util.List;
+import java.io.File;
+import java.util.ArrayList;
+
+public class RecentFileList {
+
+	private ArrayList<File> list;
+	private int maxSize = 15;
+
+	public RecentFileList() {
+		this.list = new ArrayList<File>();
+	}
+
+	public RecentFileList(int size) {
+		this.list = new ArrayList<File>();
+		this.maxSize = size;
+	}
+
+	public int size() {
+		return list.size();
+	}
+
+	public void openFile(File file) {
+		if (list.size() >= maxSize) {
+			list.remove(0);
+		}
+		if (!list.contains(file)) {
+			list.add(file);
+		} else {
+			list.remove(file);

```

Comment:
you could use the public method on this class to delete the file so you have the whole functionality into one algorithm